src/methods/coverage.jl

```diff
@@ -259,31 +239,75 @@ function _union_coverage!(A::AbstractRaster, geom;
+        put!(missed_pixels, 0)
+    end
+    range = axes(A, Y())
+    _run(range, threaded, progress, "Combining coverage...") do y
+        for x in axes(A, X())
+            D = (X(x), Y(y))
+            if center_covered[D...]
+                pixel_coverage = 1.0
+                A[D...] = pixel_coverage
+            else
+                pixel_coverage = 0.0
+                subxs, subys = map((x, y)) do i
+                    i1 = (i - 1) * scale
+                    sub_indices = i1 + 1:i1 + scale
+                end
+                pixel_coverage = sum(view(subpixel_raster, X(subxs), Y(subys))) / scale^2
+                if pixel_coverage == 0.0
+                    # Check if this line should be covered
+                    if line_covered[D...]
+                        m = take!(missed_pixels)
+                        put!(missed_pixels, m + 1)
+                    end
+                end
+                A[D...] = pixel_coverage
+            end
+        end
+    end
+    close(missed_pixels)
+    return sum(collect(missed_pixels))
+end
+
 # Sums all coverage 
 function _sum_coverage!(A::AbstractRaster, geoms, buffers; 
     scale, subpixel_dims, verbose=true, progress=true, threaded=false, checkmem=CHECKMEM[],
 )
     threaded, n = _check_buffer_thread_mem(A; scale, threaded, checkmem)
-    coveragebuffers = [fill!(similar(A), 0.0) for _ in 1:n]
-    missed_pixels = fill(0, n)
+    coveragebuffer = fill!(similar(A), 0.0)
+    allbuffers1 = merge(buffers, (; coveragebuffer, missed_pixels = [0]))
+    allbuffers = _maybe_channel(allbuffers1, threaded, n)
     range = _geomindices(geoms)
     burnchecks = _alloc_burnchecks(range)
     _run(range, threaded, progress, "Calculating coverage...") do i
         geom = _getgeom(geoms, i)
         ismissing(geom) && return nothing
-        idx = Threads.threadid()
-        thread_buffers = map(b -> b[idx], buffers)
-        coveragebuffer = coveragebuffers[idx]
-        nmissed, burnchecks[i] = _sum_coverage!(coveragebuffer, geom; scale, thread_buffers...)
-        missed_pixels[idx] += nmissed
-        fill!(thread_buffers.linebuffer, false)
-        fill!(thread_buffers.centerbuffer, false)
+        with_resource(allbuffers) do buffers
+            thread_buffers = Base.structdiff(buffers, NamedTuple{(:coveragebuffer, :missed_pixels)})
+            nmissed, burnchecks[i] = _sum_coverage!(buffers.coveragebuffer, geom; scale, thread_buffers...)
+            buffers.missed_pixels .+= nmissed
+            fill!(thread_buffers.linebuffer, false)
+            fill!(thread_buffers.centerbuffer, false)
+        end
         return nothing
     end
-    _do_broadcast!(+, A, coveragebuffers...)
     _set_burnchecks(burnchecks, metadata(A), verbose)
-    return sum(missed_pixels)
+    if threaded
+        close(allbuffers)
+        _buffers = collect(allbuffers)
+        _do_broadcast!(+, A, getindex.(_buffers, :coveragebuffer)...)
+        return sum(first.(getindex.(_buffers, :missed_pixels)))
```

Comment:
```suggestion
        return sum(x -> first(x.missed_pixels), _buffers)
```

To cut down on those allocations